### PR TITLE
Implement GetPgDependTuplesForDependingObjects

### PR DESCRIPTION
**Pull request by onurctirtir**
Implement `GetPgDependTuplesForDependingObjects` helper function for #4453 by refactoring `BuildViewDependencyGraph`.

### CODE REVIEW

File: src/backend/distributed/metadata/dependency.c [hunk 1]
```diff
@@ -1068,25 +1068,13 @@ BuildViewDependencyGraph(Oid relationId, HTAB *nodeMap)
 	node->remainingDependencyCount = 0;
 	node->dependingNodes = NIL;
 
-	ObjectAddress target = { 0 };
-	ObjectAddressSet(target, RelationRelationId, relationId);
+	Oid targetObjectClassId = RelationRelationId;
+	Oid targetObjectId = relationId;
+	List *dependencyTupleList = GetPgDependTuplesForDependingObjects(targetObjectClassId,
```
Comment (onurctirtir) [inline src/backend/distributed/metadata/dependency.c:1073]: should free tuples in the end (`heap_freetuple`)

Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1073]: maybe comment this in `GetPgDependTuplesForDependingObjects` function

File: src/backend/distributed/metadata/dependency.c [hunk 2]
```diff
@@ -1101,10 +1089,45 @@ BuildViewDependencyGraph(Oid relationId, HTAB *nodeMap)
...
+	return node;
+}
+
+
+/*
+ * GetPgDependTuplesForDependingObjects scans pg_depend for given object and
+ * returns a list of heap tuples for the objects depending on it.
+ */
+List *
+GetPgDependTuplesForDependingObjects(Oid targetObjectClassId, Oid targetObjectId)
```
Comment (halilozanakgul) [inline src/backend/distributed/metadata/dependency.c:1101]: Can you choose a different class for an object?

Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1101]: I couldn't find a way to determine class of an object given the oid, do you know a way to know this ?

File: src/backend/distributed/metadata/dependency.c (hunk 1 above)
Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1073]: 71cc4aa

Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1073]: changed my mind, actually it's not even needed

File: src/backend/distributed/metadata/dependency.c [hunk 3]
```diff
@@ -1068,25 +1068,13 @@ BuildViewDependencyGraph(Oid relationId, HTAB *nodeMap)
 	node->remainingDependencyCount = 0;
 	node->dependingNodes = NIL;
 
-	ObjectAddress target = { 0 };
-	ObjectAddressSet(target, RelationRelationId, relationId);
```
Comment (halilozanakgul) [inline src/backend/distributed/metadata/dependency.c:1072]: I wonder the purpose of doing this 😄

**Review by halilozanakgul — APPROVED** [review]

**Pull request merged**